monai/utils/enums.py

```diff
@@ -647,6 +648,37 @@ class LazyAttr(StrEnum):
     RESAMPLE_MODE = "lazy_resample_mode"
 
 
+class LazyMode(StrEnum):
+    """
+    Lazy evaluation modes for executing processing pipelines (ie. Compose). These modes control how transforms
+    that can execute lazily are executed by the pipeline:
+    'OFF' indicates that the pipeline should not be executed lazily
+    'ENABLED' indicates that the pipeline can be executed lazily, but this will only be done for transforms
+    that have ``lazy`` set to True
+    'ON' indicates that all transforms capable of being executed lazily will be executed lazily
+    See: :py:class: monai.transforms.compose.Compose for more details.
+    """
+
+    OFF = "off"
+    ENABLED = "enabled"
+    ON = "on"
+
+    @staticmethod
+    def as_bool(lazy_mode):
```

Comment:
this `as_bool` doesn't actually return a boolean value, not sure what the intention is. maybe override `self.__bool__` instead of creating this method.

Reply:
To be honest, I thought there would be more value to having LazyMode as a enum than there turns out to be. I think we could drop it and stay with bool | None. What do you think?

Reply:
It is to be able to map between lazy modes and an optional bool. As such __bool__ won't work, so I did as_bool instead

Reply:
I feel the possible values of input argument `lazy` could be `True` `False` `None` `LazyMode.OFF`, `LazyMode.ON`, `LazyMode.ENABLED`, it's not easy to use

Reply:
So, just True, False, None then?

Reply:
sure, sounds good